opshin/optimize/optimize_union_expansion.py

```diff
@@ -0,0 +1,220 @@
+from _ast import BoolOp, Call, FunctionDef, If, UnaryOp
+from ast import *
+from typing import Any, List
+from ..util import CompilingNodeTransformer
+from copy import deepcopy
+
+"""
+Expand union types
+"""
+
+
+def type_to_suffix(typ: expr) -> str:
+    try:
+        raw = unparse(typ)
+    except Exception:
+        return "UnknownType"
+    return (
```

Comment:
This expansion should make sure that we can not accidentally create the same function name manually and then accidentally overwrite it. One way to ensure this would be to also escape underscores to "___". Additionally we should check that the generated name was not chosen manually before (or ensure it it by e.g. using a number in front)

Reply:
I think even with prefixing 0, we can get overlaps for example in `foo(x: Union[int, bytes], y: Union[int, bytes])` and foo_int(y: Union[int, bytes]`

Reply:
Currently a collision would require something like this:

`foo(x: Union[int, bytes], y: Union[int, bytes]) `and `foo_eut_int(y: Union[int, bytes])` as I also add the suffix `"_eut"` 

I could make the suffix more complicated or add an illegal symbol   "_+eut"

Reply:
can you just use + as a seperator?

Reply:
Yeah, that's fine. No need for a number prefix either then.